cmd/spx/template/game.js

```diff
@@ -118,14 +128,37 @@ class GameApp {
         deleteInfos = deleteInfos.map(info => `res://${info}`);
         const evt = new CustomEvent('update_project', {
             detail: {
-                "resolve": resolve,
+                "resolve": async () => {
+                    await this.checkAndUpdateCache(newData)
+                    resolve()
+                },
                 "dirtyInfos": datas,
                 "deleteInfos": deleteInfos,
             }
         });
         this.editorCanvas.dispatchEvent(evt);
     }
 
+    async stopProject(resolve, reject) {
+        console.log("stop project")
+        if (this.editor == null) {
+            resolve()
+            return
+        }
+        this.stopGameTask++
+        await this.stopGame(() => {
+            this.isEditor = true
```

Comment:
What does `isEditor` mean? 

The current logic indicates it is equivalent to `!isRunningGame`. Is that expected?

Reply:
This is a tag used to indicate whether the focus should be on the game canvas or the editor canvas.

Reply:
When should I call `StopProject` in Builder?

Reply:
When the user wants to switch to another project

Reply:
OK